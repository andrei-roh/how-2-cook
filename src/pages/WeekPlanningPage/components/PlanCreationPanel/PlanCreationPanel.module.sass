@use 'src/style/colors'
@use 'src/style/breakpoints'
@import 'src/style/typography'

div .createPlanButton
    @extend .H6
    width: 250px
    height: 48px
    box-sizing: border-box
    color: colors.$white
    background-color: colors.$blue
    margin-bottom: 32px

    &:hover
        background-color: colors.$sky

    &:disabled,
    &[disabled]
        cursor: auto 
        background-color: colors.$gray


div .creationPanel
    width: calc(100vw - 48px)
    box-sizing: border-box

    .creationButton
        width: 110px
        background-color: colors.$blue
        color: colors.$white

        &:hover
            background-color: colors.$sky
        &:disabled,
        &[disabled]
            background-color: colors.$gray