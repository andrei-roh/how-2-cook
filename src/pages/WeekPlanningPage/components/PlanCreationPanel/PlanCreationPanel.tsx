import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';
import css from './PlanCreationPanel.module.sass';

interface PlanCreationPanelProps {
  isStartCreation: boolean;
  isAllocatedRecipes: boolean;
  planName: string;
  setPlanName: React.Dispatch<React.SetStateAction<string>>;
  handleCreatePlan: () => void;
  startCreationPlanningList: () => void;
}

export const PlanCreationPanel = ({
  isStartCreation,
  isAllocatedRecipes,
  planName,
  setPlanName,
  handleCreatePlan,
  startCreationPlanningList,
}: PlanCreationPanelProps) => {
  return isStartCreation ? (
    <TextField
      label='Создание плана'
      fullWidth
      value={planName}
      onChange={({ target: { value } }) => setPlanName(value)}
      className={css.creationPanel}
      InputProps={{
        endAdornment: (
          <Button
            variant='contained'
            className={css.creationButton}
            onClick={handleCreatePlan}
            disabled={!planName || !isAllocatedRecipes}
          >
            Создать
          </Button>
        ),
      }}
    />
  ) : (
    <Button
      className={css.createPlanButton}
      onClick={startCreationPlanningList}
    >
      Создать список рецептов
    </Button>
  );
};
