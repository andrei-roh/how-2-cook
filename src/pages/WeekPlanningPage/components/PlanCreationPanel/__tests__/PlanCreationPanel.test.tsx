import { describe, expect, it, vi } from 'vitest';
import { render } from '@testing-library/react';
import { PlanCreationPanel } from '../PlanCreationPanel';

const planNameMock = 'Test Plan';
const setPlanNameMock = vi.fn();
const handleCreatePlanMock = vi.fn();
const startCreationPlanningListMock = vi.fn();

describe('Testing: PlanCreationPanel', () => {
  it('should render when isStartCreation is true', () => {
    const { container } = render(
      <PlanCreationPanel
        isStartCreation={true}
        isAllocatedRecipes={true}
        planName={planNameMock}
        setPlanName={setPlanNameMock}
        handleCreatePlan={handleCreatePlanMock}
        startCreationPlanningList={startCreationPlanningListMock}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render when isStartCreation is false', () => {
    const { container } = render(
      <PlanCreationPanel
        isStartCreation={false}
        isAllocatedRecipes={true}
        planName={planNameMock}
        setPlanName={setPlanNameMock}
        handleCreatePlan={handleCreatePlanMock}
        startCreationPlanningList={startCreationPlanningListMock}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
