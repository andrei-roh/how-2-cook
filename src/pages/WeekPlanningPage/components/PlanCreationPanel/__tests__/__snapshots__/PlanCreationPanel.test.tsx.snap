// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: PlanCreationPanel > should render when isStartCreation is false 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary _createPlanButton_b07739 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Создать список рецептов
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Testing: PlanCreationPanel > should render when isStartCreation is true 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root _creationPanel_b07739 css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r0:"
      id=":r0:-label"
    >
      Создание плана
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        type="text"
        value="Test Plan"
      />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary _creationButton_b07739 css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Создать
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Создание плана
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
